Functions/Mock.ps1

```diff
@@ -1659,7 +1659,12 @@ function New-BlockWithoutParameterAliases {
     try {
         if ($PSVersionTable.PSVersion.Major -ge 3) {
             $params = $Metadata.Parameters.Values
-            $ast = $Block.Ast.EndBlock
+            if ($block.Ast -is [System.Management.Automation.Language.FunctionDefinitionAst]) {
+                $ast = $block.Ast.Body.EndBlock
+            }
+            elseif ($block.Ast -is [System.Management.Automation.Language.ScriptBlockAst]) {
```

Comment:
What happens if it is neither? 

Please use same capitalization for `Block` with capital `B` :)

Reply:
I have been reading about conversions to scriptblocks, but I think this should give us enough coverage. I was worried that .Net delegates could be transformed into scriptblocks with a different AST structure, but it does not seem to be the case

Reply:
Okay, so if there is no third option then it should throw on `else`?

Reply:
Thanks for the quick feedback!

I've changed the casing on to match: $Block .

I also added a throw for the case of it getting through the function param validation [scriptblock] , but not having a type of 'FunctionDefinitionAst' or 'ScriptBlockAst' . This probably won't ever happen - but just in case.